src/main/java/dev/crec/mapinslot/MapInSlot.java

```diff
@@ -9,19 +9,27 @@
 import net.minecraft.world.level.saveddata.maps.MapItemSavedData;
 
 public class MapInSlot {
-	public static void renderMap(Minecraft mc, float blitOffset, ItemStack stack, int x, int y) {
+	public static void renderMap(PoseStack poseStack1, Minecraft mc, float blitOffset, ItemStack stack, int x, int y) {
 		if (mc.player == null) return;
 
 		Integer mapId = MapItem.getMapId(stack);
 		MapItemSavedData saveData = MapItem.getSavedData(mapId, mc.player.level);
 
 		if (mapId != null && saveData != null) {
 			MultiBufferSource.BufferSource bufferSource = MultiBufferSource.immediate(Tesselator.getInstance().getBuilder());
-			PoseStack poseStack = new PoseStack();
-			poseStack.translate(x, y, blitOffset + 190);
-			poseStack.scale(1/8F, 1/8F, 1);
-			mc.gameRenderer.getMapRenderer().render(poseStack, bufferSource, mapId, saveData, true, 0xF000D2);
-			bufferSource.endBatch();
+			if (poseStack1.clear()) {
+				PoseStack poseStack = new PoseStack();
+				poseStack.translate(x, y, blitOffset + 190);
+				poseStack.scale(1 / 8F, 1 / 8F, 1);
+				mc.gameRenderer.getMapRenderer().render(poseStack, bufferSource, mapId, saveData, true, 0xF000D2);
+				bufferSource.endBatch();
+			} else {
+				PoseStack poseStack = poseStack1;
+				poseStack.translate(x, y, blitOffset + 190);
+				poseStack.scale(1 / 8F, 1 / 8F, 1);
+				mc.gameRenderer.getMapRenderer().render(poseStack, bufferSource, mapId, saveData, true, 0xF000D2);
+				bufferSource.endBatch();
+			}
```

Comment:
I am not sure what PoseStack#clear() does (I will do a through second review later), but instead of code duplication, you can simply check the clear otherwise reassign the variable to new PoseStack. This will remove code duplication and make this whole bit cleaner.